### PR TITLE
Writes Kubernetes logs files in UTF-8

### PR DESCRIPTION
**Pull request by pcm32**
To avoid exceptions produced when trying to write non-ascii characters present in logs output.

### CODE REVIEW

File: lib/galaxy/jobs/runners/kubernetes.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def __produce_log_file(self, job_state):
             logs += "\n\n==== Pod " + pod.name + " log end   ===="
         logs_file_path = job_state.output_file
         logs_file = open(logs_file_path, mode="w")
-        logs_file.write(logs)
+        logs_file.write(logs.encode('utf8'))
```
Comment (nsoranzo) [inline lib/galaxy/jobs/runners/kubernetes.py]: I think this is safer:

```
from six import text_type
...

        if isinstance(logs, text_type):
            logs = logs.encode('utf8')
        logs_file.write(logs)
```

Comment (pcm32) [inline lib/galaxy/jobs/runners/kubernetes.py]: Thanks @nsoranzo, I have implemented this suggestion now.

**Comment by nsoranzo** [comment]
:+1:

**Comment by nsoranzo** [comment]
@galaxybot test this

**Pull request merged**